File: backend/src/main/kotlin/fr/gouv/cnsp/monitorfish/config/SwaggerConfig.java
package fr.gouv.cnsp.monitorfish.config;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import springfox.documentation.builders.PathSelectors;
import springfox.documentation.builders.RequestHandlerSelectors;
import springfox.documentation.service.ApiInfo;
import springfox.documentation.service.Contact;
import springfox.documentation.spi.DocumentationType;
import springfox.documentation.spring.web.plugins.Docket;
import springfox.documentation.swagger2.annotations.EnableSwagger2;

import java.util.Collections;

@Configuration
@EnableSwagger2
public class SwaggerConfig {

    @Autowired
    private HostProperties hostProperties;

    @Bean
    public Docket airQualityApi() {
        return new Docket(DocumentationType.SWAGGER_2)
                .host(hostProperties.getIp())
                .select()
                .apis(RequestHandlerSelectors.basePackage("fr.gouv.cnsp.monitorfish"))
                .paths(PathSelectors.any())
                .build()
                .apiInfo(new ApiInfo(
                        "API Documentation",
                        "This is public API",
                        "v1",
                        "",
                        new Contact("CNSP", "URL", "EMAIL"),
                        "APACHE",
                        "LICENSE URL",
                        Collections.emptyList()
                ))
                .enableUrlTemplating(true);
    }
}

File: backend/src/main/kotlin/fr/gouv/cnsp/monitorfish/config/SwaggerConfig.kt
package fr.gouv.cnsp.monitorfish.config

import io.swagger.v3.oas.models.ExternalDocumentation
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import io.swagger.v3.oas.models.info.License
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class SwaggerConfig {
    @Autowired
    private val hostProperties: HostProperties? = null

    @Bean
    fun api(): OpenAPI {
        return OpenAPI()
            .info(
                Info().title("MonitorFish API")
                    .description("MonitorFish")
                    .version("v1.19.2")
                    .license(License().name("Apache 2.0").url("https://monitorfish.readthedocs.io/en/latest")),
            )
            .externalDocs(
                ExternalDocumentation()
                    .description("MonitorFish Documentation")
                    .url("https://monitorfish.readthedocs.io/en/latest"),
            )
    }
}
